src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs

```diff
@@ -1181,6 +1360,50 @@ class Test
             await VerifyItemExistsAsync(markup, "tokens");
         }
 
+        [WorkItem(37366, "https://github.com/dotnet/roslyn/issues/37366")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task PluralizeSpan()
+        {
+            var markup = @"
+using System;
+
+class Test
+{
+    void M(Span<Test> $$) { }
+}
+" + Span;
+            await VerifyItemExistsAsync(markup, "tests");
+        }
+
+        [WorkItem(37366, "https://github.com/dotnet/roslyn/issues/37366")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task PluralizeValidGetEnumerator()
+        {
+            var markup = @"
+class MyClass
+{
+    public void M(MyOwnCollection<MyClass> $$) { }
+}
+
+
+class MyOwnCollection<T>
+{
+    public MyEnumerator GetEnumerator()
+    {
+        return new MyEnumerator();
+    }
+
+    public class MyEnumerator
+    {
+        public T Current { get; }
+        
+        public bool MoveNext() { return false; }
+    }
+}
+";
+            await VerifyItemExistsAsync(markup, "myClasses");
+        }
```

Comment:
i don't see tests for:

1. valid GetAsyncEnumerator
2. not finding GetEnumerator and looking to see if an interface is implemented.

Reply:
Sure, I'll add more tests.